### PR TITLE
feat(tax): add endpoints to create tax regions and tax rates

### PR DESCRIPTION
**Pull request by srindom**
**What**
Adds:
- POST /admin/tax-regions
- POST /admin/tax-rates
- GET /admin/tax-rates
- `createTaxRegionsWorkflow`
- `createTaxRatesWorkflow`

### CODE REVIEW

File: packages/medusa/src/api-v2/admin/tax-regions/validators.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+}) {
+  /**
+   * Search parameter for regions.
+   */
+  @IsString({ each: true })
+  @IsOptional()
+  id?: string | string[]
+
+  /**
+   * Filter by currency code
```
Comment (sradevski) [inline packages/medusa/src/api-v2/admin/tax-regions/validators.ts]: The comment here and below are incorrect 🙈

File: packages/medusa/src/api-v2/admin/tax-regions/validators.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  $or?: AdminGetTaxRegionsParams[]
+}
+
+class CreateDefaultTaxRate {
+  @IsOptional()
+  @IsType([Number])
+  rate?: number | null
+
+  @IsOptional()
+  @IsType([String])
```
Comment (sradevski) [inline packages/medusa/src/api-v2/admin/tax-regions/validators.ts]:
```suggestion
@IsString
```

File: packages/medusa/src/api-v2/admin/tax-regions/validators.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+  @IsOptional()
+  @ValidateNested({ each: true })
+  @Type(() => AdminGetTaxRegionsParams)
+  $or?: AdminGetTaxRegionsParams[]
+}
+
+class CreateDefaultTaxRate {
+  @IsOptional()
+  @IsType([Number])
```
Comment (sradevski) [inline packages/medusa/src/api-v2/admin/tax-regions/validators.ts]:
```suggestion
@IsNumber
```

File: packages/types/src/tax/common.ts [hunk 4]
```diff
@@ -22,6 +22,18 @@ export interface TaxRateDTO {
...
    */
   metadata: Record<string, unknown> | null
+  /**
+   * The id of the Tax Region the rate is associated with.
+   */
+  tax_region_id: string
+  /**
+   * Flag to indicate if the Tax Rate should be combined with parent rates.
+   */
+  is_combinable: boolean
```
Comment (sradevski) [inline packages/types/src/tax/common.ts:32]: I am completely clueless on this one, what does it mean to be able to combine a tax rate with a parent rate? If this is explained somewhere in Notion feel free to just share a link to that 😄

File: packages/medusa/src/api-v2/admin/tax-rates/route.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+export const GET = async (
+  req: AuthenticatedMedusaRequest,
+  res: MedusaResponse
+) => {
+  const remoteQuery = req.scope.resolve("remoteQuery")
+
+  const variables = { id: req.params.id }
+
+  const queryObject = remoteQueryObjectFromString({
+    entryPoint: "tax_rates",
```
Comment (sradevski) [inline packages/medusa/src/api-v2/admin/tax-rates/route.ts]: We have it as plural here and singular above, I think using just one for consistency is better (I assume due to the aliases this works as well, right?)

File: packages/core-flows/src/tax/steps/create-tax-regions.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      created.map((region) => region.id)
+    )
+  },
+  async (createdIds, { container }) => {
+    if (!createdIds?.length) {
+      return
+    }
+
+    const service = container.resolve<ITaxModuleService>(
+      ModuleRegistrationName.REGION
```
Comment (sradevski) [inline packages/core-flows/src/tax/steps/create-tax-regions.ts]:
```suggestion
      ModuleRegistrationName.TAX
```

File: packages/types/src/tax/common.ts (hunk 4 above)
Comment (srindom) [reply packages/types/src/tax/common.ts:32]: In some countries, multiple authorities can charge tax. For example, in Canada, you pay a federal sales tax and a provincial sales tax. 

The way that would be set up would be something like:
- COUNTRY Region
  - Tax Rate: X
  - PROVINCE Region
    - parent_id: state region id
    - Tax Rate: Y

The is_combinable flag indicates whether the end customer should be charged X+Y or just Y. 

Does this make more sense?

Comment (sradevski) [reply packages/types/src/tax/common.ts:32]: Yeah makes perfect sense, thanks @srindom !

**Pull request merged**